### ISSUE TITLE
PlaceRadius.contains_edge

### ISSUE THREAD

**Issue by yakra** [TravelMapping/DataProcessing#192]
Our list of `matching_vertices` is already known.
Rather than perform all the trig functions and compare distance to the radius, it *may* be faster to check both points for membership in the list.

**Comment by yakra** [comment]
The changes for #187 will change `matching_vertices` from a list to a set. This is a Good Thing, as checking for membership in a set is faster then a list.

**Comment by yakra** [comment]
C++: Area graph generation took 12.8s, compared to 12.6 [before](https://github.com/TravelMapping/DataProcessing/issues/187#issuecomment-462162815).
Still worth giving this a try in Python; it's a simple enough modification.

**Comment by yakra** [comment]
Python: Area graph generation took 49.5s, compared to 50.2 [before](https://github.com/TravelMapping/DataProcessing/issues/187#issuecomment-462236303). This is either a small decrease in processing time, or within margin of error.
I'll run some more tests to try to get a better handle on which is the case.

**Comment by yakra** [comment]
Looks like we're faster by about half a second:
```
math		50.8	50.6	50.4	50.3	50.2
membership	50.8	50.3	49.9	49.5	49.3
```
Another idea that may be faster than checking set membership:
When adding vertices to a new per-PlaceRadius set, use the `vertex_num` variable to flag it as part of the set. Check `vertex_num` when `matching_edges`.

**Comment by yakra** [comment]
```
Area Graphs
====================================================
math		50.8	50.6	50.4	50.3	50.2
vertex_num	49.3	48.7	48.5	47.8	47.2


Setup
====================================================
math		26.7	26.2	26.1	25.9	25.8
vertex_num	27.8	27.4	27.4	26.6	26.3
```
Heh. So we're saving 0.9-3.4s, 2.2s on average.
And possibly taking up about 1s more during setup, due to initializing two lousy variables here, though this is getting to be within margin of error. For the sake of Jim's sanity, I'm not going to bother opening a pull request right away, but here's the code: https://github.com/yakra/DataProcessing/commit/622da4f885fb1099a3e0f4ce2eb8580ee6b1c5d8

**Comment by yakra** [comment]
Back to C++ now, looking at the `vertex_num = -1` method...
```
Setup
========================================================
math		11.9	11.9	11.9	11.9	11.8
vertex_num	11.9	11.9	11.9	11.9	11.9

Area Graphs
========================================================
math		13.0	12.8	12.8	12.8	12.8
vertex_num	12.9	12.7	12.7	12.6	12.6
```
Setup, we lose ~0.02s.
Area graph gen, we gain about ~0.14s.

Once multithreading enters the picture:
* The area graph gen time savings will be cut in, I'll estimate, about 3, in the case of 4 threads.
* Setup (which I don't plan to multithread) time will be about the same, possibly increasing a smidge since `vertex_num` will have to be initialized for each thread. Yeah, `for` loops...

Keeping it simple & not implementing this.

**Comment by yakra** [comment]
And with that, it's the end of the road for my attempts at speeding up graph generation in siteupdate.py.

**Comment by jteresco** [comment]
And what an excellent road it was.  Thanks for all of that!

I am going to wait until my current set of lab assignments is done before bringing in recent changes and then regenerating graphs.  Today's snow-out means that what should have been substantially done tomorrow will now stretch into late next week.

**Comment by yakra** [comment]
Till then, what we already have is pretty good.

And yowch. We got hit by that here in Portland too. Heavy, immovable grease that mires cars and clogs snowblowers. It's been fun.

**Comment by yakra** [comment]
notes on http://yakra.teresco.org/logs/pr_cont_e2/siteupdate_s1.log before deleting it...
Appears to be generated on noreaster itself rather than BiggaTomato, based on runtime of 453.9.
Retesting with
DataProcessing at 622da4f
HighwayData at c4837b1eb6
UserData at b8cd6d76
435.0 on noreaster, 604.7 on BiggaTomato
48.5s for area graphs, 26.1s for setup. I don't *think* I mixed up the categories [above](https://github.com/TravelMapping/DataProcessing/issues/192#issuecomment-462840141); this must just be a separate file from whatever it was I did for time trials (date & time = Feb 11 22:51, earlier than the above comment).
Suggests there's actually no appreciable difference to setup time when just initializing a couple variables, which makes sense.
